src/main/java/org/wildfly/security/http/impl/ClientCertAuthenticationMechanism.java

```diff
@@ -86,23 +86,22 @@ public void evaluateRequest(HttpServerRequest request) throws HttpAuthentication
             log.trace("ClientCertAuthenticationMechanism: re-authentication succeed");
             return;
         }
-        if (attemptAuthentication(request, sslSession)) {
-            log.trace("ClientCertAuthenticationMechanism: authentication succeed");
+        if (attemptAuthentication(request)) {
             return;
         }
         log.trace("ClientCertAuthenticationMechanism: both, re-authentication and authentication, failed");
         fail(request);
     }
 
-    private boolean attemptAuthentication(HttpServerRequest request, SSLSession sslSession) throws HttpAuthenticationException {
-        X509Certificate[] x509Certificates;
-        try {
-            x509Certificates = X500.asX509CertificateArray(sslSession.getPeerCertificates());
-        } catch (SSLPeerUnverifiedException e) {
+    private boolean attemptAuthentication(HttpServerRequest request) throws HttpAuthenticationException {
+        Certificate[] peerCertificates = request.getPeerCertificates();
+        if (peerCertificates == null) {
```

Comment:
Does renegotiation happens here? I still wonder which UC does it solve? How can that happen that CLIENT_CERT SSLSession was created without peer certificate?